File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    queue = deque([(p, i) for i, p in enumerate(priorities)])
    order = 0

    while queue:
        priority, idx = queue.popleft()

        # 더 높은 우선순위가 있는지 확인
        if any(priority < q[0] for q in queue):
            queue.append((priority, idx))
        else:
            order += 1
            if idx == location:
                return order

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let queue = priorities.map((p, idx) => [p, idx]);
    let order = 0;

    while (queue.length > 0) {
        const current = queue.shift();

        // 큐에 더 높은 우선순위가 있는지 확인
        if (queue.some(item => item[0] > current[0])) {
            queue.push(current);
        } else {
            // 실행
            order++;
            if (current[1] === location) {
                return order;
            }
        }
    }
}
